cvm/auth-service/src/auth_service/main.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import os
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+logging.basicConfig(
+    level=os.environ.get("LOG_LEVEL", "INFO").upper(),
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+logger = logging.getLogger("auth-service")
+
+AUTH_TOKEN = os.environ.get("AUTH_TOKEN")
+HOST = os.environ.get("HOST", "0.0.0.0")
+PORT = int(os.environ.get("PORT", "8081"))
+
+
+class AuthHandler(BaseHTTPRequestHandler):
+    def log_message(self, format, *args):
+        logger.info("%s - %s", self.address_string(), format % args)
+
+    def do_GET(self):
+        if self.path == "/health":
+            self.send_response(200)
+            self.send_header("Content-Type", "text/plain")
+            self.end_headers()
+            self.wfile.write(b"healthy")
+            return
+
+        if self.path == "/auth":
+            auth_header = self.headers.get("Authorization", "")
+
+            if not AUTH_TOKEN:
+                logger.error("AUTH_TOKEN environment variable not set")
+                self.send_response(500)
+                self.end_headers()
+                return
+
+            if auth_header == f"Bearer {AUTH_TOKEN}":
```

Comment:
I think we should use something that is not time dependent to compare the values.

Reply:
Indeed! I'll fix that.